analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/parser/CommitCollector.kt

```diff
@@ -10,70 +11,98 @@ import java.util.stream.Collector
-        return versionControlledFiles.add(missingVersionControlledFile)
-    }
+            when (it.type) {
+                Modification.Type.ADD -> {
+                    val file: VersionControlledFile? =
+                        versionControlledFiles[possibleConflictName] //does the file exist?
+                    if (file == null) {
+                        val missingVersionControlledFile =
+                            VersionControlledFile(possibleConflictName, metricsFactory) // if not create a file
+                        versionControlledFiles[possibleConflictName] =
+                            missingVersionControlledFile // and add it to the list
+                        missingVersionControlledFile.registerCommit(commit, it)
+                    } else {
+                        versionControlledFiles[VCFName]!!.registerCommit(commit, it)
+                    }
+                }
+                Modification.Type.DELETE -> {
+                    val filename = renamesMap[possibleConflictName]
+                    versionControlledFiles.remove(if (filename == null) possibleConflictName else filename) //remove a deleted file from list
+                    renamesMap.remove(possibleConflictName) // and remove its references in renames
+                }
+                Modification.Type.RENAME -> {
+
+                    var newVCFFileName = it.currentFilename
+
+                    if (versionControlledFiles.containsKey(it.currentFilename)) {//is the filename contained in vCF
+                        val marker: Int? =
+                            nameConflictsMap[it.currentFilename] //check if the file is already contained with a marker
+                        val newMarker: Int =
+                            if (marker != null) marker + 1 else 0 //increment the marker for additional tracking
 
-    private fun addCommitMetadataToMatchingVersionControlledFiles(
-        commit: Commit,
-        versionControlledFiles: List<VersionControlledFile>
-    ) {
-        commit.filenames.mapNotNull { findVersionControlledFileByFilename(versionControlledFiles, it) }
-                .forEach { it.registerCommit(commit) }
+                        newVCFFileName =
+                            it.currentFilename + "_\\0_" + newMarker // generate new vCF entry we work with it.currentFilename to preserve the string structure
+                    }
+                    if (renameName != null) {
+                        renamesMap.remove(possibleConflictName) //remove old entry
+                        renamesMap[newVCFFileName] = renameName // insert new entry, tracking the current name
+                    } else {
+                        renamesMap[newVCFFileName] = it.oldFilename //add new entry
+                    }
+                    versionControlledFiles[VCFName]?.registerCommit(commit, it)
+                }
+
+                else -> versionControlledFiles[VCFName]?.registerCommit(commit, it)
```

Comment:
```suggestion
                else -> versionControlledFiles[VCFName]!!.registerCommit(commit, it)
```

Reply:
That wont work, will lead to a NullPointer

Reply:
That indicates that there's an error in the logic somewhere. This should be looked at closely.